modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -3920,10 +3924,10 @@ public void walFilesDeleted(int walFilesDeleted) {
         }
 
         /**
-         * @param walSegmentsCovered WAL segments fully covered by this checkpoint.
+         * @param walSegsCoveredRange WAL segments fully covered by this checkpoint.
          */
-        public void walSegmentsCovered(final List<Long> walSegmentsCovered) {
-            this.walSegmentsCovered = walSegmentsCovered;
+        public void walSegsCoveredRange(final IgniteBiTuple<Long, Long> walSegsCoveredRange) {
```

Comment:
I think it can be sipmlified to `(final long begin, final long end)`

Reply:
It is questionable: from point of view of method signature it is simpler but more code should be added to Checkpoint entity: getters and setters for two fields instead of one.

Also BiTuple represents here a concept of "range" while two separate parameters don't capture it as good as single one.